Source/JavaScriptCore/runtime/JSFunction.h

```diff
@@ -174,6 +174,16 @@ class JSFunction : public JSCallee {
 
     bool mayHaveNonReifiedPrototype();
 
+    // This method may be called for host functions, in which case it
+    // will return an arbitrary value. This should only be used for
+    // optimized paths in which the return value does not matter for
+    // host functions, and checking whether the function is a host
+    // function is deemed too expensive.
+    JSScope* scopeUnchecked()
+    {
+        return m_scope.get();
+    }
```

Comment:
The motivation of moving `scopeUnchecked` from `JSCallee` to `JSFunction`?

Reply:
We would like to limit the function touching this scope to a limited class. So limit scopeUnchecked use only for JSFunction, not for JSCallee.